### DISCUSSION TITLE
Error when trying to set maximumCrawlCount

### DISCUSSION THREAD

**Discussion by goodjobs-ta** [spatie/laravel-sitemap#485] [Q&A]
Hi,
when I try to set the maximumCrawlCount value and create a sitemap afterwards I run into the following error:

 Call to undefined method Spatie\Crawler\Crawler::setMaximumCrawlCount()
 at vendor/spatie/laravel-sitemap/src/SitemapGenerator.php:117

The function 'setMaximumCrawlCount' seems to be missing on the 'Spatie\Crawler\Crawler'.
I think It must be used  'setTotalCrawlLimit' instead.

**Comment by jadersbr** [comment]
Did you resolve this error? Same error here, but acordant to documentation, we needs to use ```use Spatie\Sitemap\SitemapGenerator;``` and inside this class has ```setMaximumCrawlCount```. I got this error message ```Call to undefined method Spatie\Crawler\Crawler::setMaximumCrawlCount()```.

**Comment by BoGnY** [comment]
Yeah it's a bug..
I've resolved extending `SitemapGenerator` class and overwriting `getSitemap()` function with my personal `App\Support\SitemapGenerator` and changing this:
```php
        if (! is_null($this->maximumCrawlCount)) {
            $this->crawler->setMaximumCrawlCount($this->maximumCrawlCount);
        }
```
With this:
```php
        if (! is_null($this->maximumCrawlCount)) {
            $this->crawler->setTotalCrawlLimit($this->maximumCrawlCount);
        }
```

Then instead using `Spatie\Sitemap\SitemapGenerator` just use `App\Support\SitemapGenerator`.

Cheers 👍🏻

**Comment by goodjobs-ta** [comment]
Ok, thanks so far. So if this a bug, where to open an issue for it?!

**Reply by BoGnY** [reply]
I create PR on #490 👍🏻

**Reply by goodjobs-ta** [reply]
@BoGnY Thanks!

**Reply by BoGnY** [reply]
PR merged, fixed in version 6.2.5 👍🏻

**Reply by goodjobs-ta** [reply]
Great news!
